### PR TITLE
Akka.Remote.ProtocolTransport: HandleMsg used instead of AssociationHandle

### LINKED ISSUE #2752: Akka.Remote: serialize-messages
When I enable all message serialization with "akka.actor.serialize-messages = on"
it's failing when trying to serialize/deserialize TcpAssociationHandle.
Shouldn't some messages in Remoting be marked with INoSerializationVerificationNeeded?
Probably yes since it's the case in Akka. Will create PR for this

### PR DESCRIPTION
**Pull request by zbynek001**
fixes #2752

### CODE REVIEW

**Comment by Danthar** [comment]
Assigned @Aaronontheweb but changed my mind. This looks ok.

**Comment by Aaronontheweb** [comment]
@Danthar actually, I'd like to take a look at this

File: src/core/Akka.Remote/Transport/AkkaProtocolTransport.cs [hunk 1]
```diff
@@ -864,15 +864,15 @@ private void InitializeFSM()
                             ou.StatusCompletionSource.SetException(f.Cause);
                             nextState = Stop();
                         }))
-                    .With<AssociationHandle>(h => fsmEvent.StateData.Match()
+                    .With<HandleMsg>(h => fsmEvent.StateData.Match()
```
Comment (Aaronontheweb) [inline src/core/Akka.Remote/Transport/AkkaProtocolTransport.cs]: Right on - this is correct. The JVM uses a case class here.

**Comment by Aaronontheweb** [comment]
This is ready for merge - changes are solid

**Pull request merged**